### PR TITLE
Feature/book event calendly/adhithya

### LINKED ISSUE #53: Book an event with Calendly

### PR DESCRIPTION
**Pull request by adhi0331**
### Tracking Info

Resolves #53 #93 

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

Opens up a safari instance with calendly link. 
Allows users to book a session via calendly.
Update Backend with session data

### Testing

How did you confirm your changes worked? 

I ran the backend server and also simulated the program using the ios simulator.

### Confirmation of Change 

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Screenshot 2023-04-27 at 9 55 09 PM
Screenshot 2023-04-27 at 9 55 36 PM

### CODE REVIEW

File: ALUM/ALUM/Services/PostSessionService.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Services/PostSessionService.swift:26]: Let's rename this to a SessionService that way it can be added to the session service added by https://github.com/TritonSE/ALUM-Mobile-Application/pull/103/files#diff-2f9a2ce64c523b70bf8358e75454965fbf08561052258bbb2911536e7b14c416

File: ALUM/ALUM/Services/PostSessionService.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Services/PostSessionService.swift:36]: Good idea for testing but let's integrate this with the auth token stuff before merge

File: ALUM/ALUM/Views/CalendlyBooking/CalendlyBooking.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Views/CalendlyBooking/CalendlyBooking.swift:28]: We can integrate this with the ProfileView now.

File: ALUM/ALUM/Views/CalendlyBooking/CalendlyBooking.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Views/CalendlyBooking/CalendlyBooking.swift:32]: We should remove the need for passing IDs to POST session cauz we are sending auth token anyways. In backend, use auth token to fill in the menteeID and then use the pairing schema to get the mentorID

https://github.com/TritonSE/ALUM-Mobile-Application/issues/93

File: backend/src/app.ts
Comment (AmanKAggarwal) [inline backend/src/app.ts:29]: What's this for?

File: backend/src/models/mentor.ts
Comment (AmanKAggarwal) [inline backend/src/models/mentor.ts:39]: Great! We also need to add the personalAccessToken to the signup page for now. 

Use this PR as reference cauz that adds 2 fields to signup flow - 
https://github.com/TritonSE/ALUM-Mobile-Application/pull/114

File: backend/src/routes/calendlyPage.ts
Comment (AmanKAggarwal) [inline backend/src/routes/calendlyPage.ts:16]: can we cleanup this HTML file? these console logs don't need to stay

File: backend/src/routes/calendlyPage.ts
Comment (AmanKAggarwal) [inline backend/src/routes/calendlyPage.ts:17]: Let's make this URL a constant as well and after deployment thsi needs to be set to the URL for our backend

File: backend/src/routes/sessions.ts
Comment (AmanKAggarwal) [inline backend/src/routes/sessions.ts:42]: We can just use `req.body.uid` to get the menteeID cauz its a mentee that would always access this page. Then use the pairing schema to get the `mentorID`

File: backend/src/services/calendly.ts
Comment (AmanKAggarwal) [inline backend/src/services/calendly.ts:34]: unused code, intentional?

File: backend/views/index.pug
Comment (AmanKAggarwal) [inline backend/views/index.pug:1]: curious, what's this for?

**Review by AmanKAggarwal — CHANGES_REQUESTED** [review]
Great work with the calendly stuff!

File: backend/src/app.ts
Comment (adhi0331) [reply backend/src/app.ts:29]: This is so it can render the pug file that is being used to template the Javascript string which is the html file.

File: backend/views/index.pug
Comment (adhi0331) [reply backend/views/index.pug:1]: https://expressjs.com/en/guide/using-template-engines.html

This link that Philip sent explains how the file templating works. Essentially it will replace calendlyPage with the actual JS string which holds the html and render it.

Comment (AmanKAggarwal) [reply backend/views/index.pug:1]: I see but right now we aren't doing templating so do you plan to add that? I defo think that'll be great because it will make for a nicer codebase

File: backend/src/routes/calendlyPage.ts
Comment (AmanKAggarwal) [inline backend/src/routes/calendlyPage.ts:59]: Ohh I see what you did here. When we use a templating engine its great because then you can just have your html file (in this case pug file) being stored as a file and not as a string like we have right now. 

Instead of passing the whole text `calendlyPage` you should create a pug file with the contents of `calendlyPage` that are not changing. And then pass in the calendlyLink while rendering

Your call would look like this - 
`res.render('index', { calendlyLink });`

Comment (AmanKAggarwal) [reply backend/src/routes/calendlyPage.ts:59]: Check this out - 
https://sharegpt.com/c/SgkuIxo

File: backend/views/index.pug
Comment (AmanKAggarwal) [reply backend/views/index.pug:1]: gotcha! if we are using a templating engine, lets make full use of it!

File: backend/src/services/calendly.ts
Comment (AmanKAggarwal) [inline backend/src/services/calendly.ts:10]: great security

File: backend/src/services/calendly.ts
Comment (AmanKAggarwal) [inline backend/src/services/calendly.ts:8]: awesomeeee! We will be able to reuse this function for Reschedule flow #115

File: ALUM/ALUM/ALUMApp.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/ALUMApp.swift:31]: we can get rid of this change right? Think this was from when we tried adding the URL scheme and stuff

File: ALUM/ALUM/Info.plist
Comment (AmanKAggarwal) [inline ALUM/ALUM/Info.plist:15]: this also

File: backend/src/routes/calendlyPage.ts
Comment (AmanKAggarwal) [inline backend/src/routes/calendlyPage.ts:5]: Let's remove the calendly_link query argument from the route. Once we add auth to this, your route should get the mentor associated with this mentee through the pairing schema (honestly getting a mentee's mentor is a very common functionality so might be useful to put in a function if not already). Once this route has the mentorId, just use that to get the calendly_link for the mentor and populate this HTML file. 

This will make it simpler and nicer during integration. GOLDEN RULE - Always try to keep as much logic out of frontend and inside backend

**Review by christen03 — APPROVED** [review]
w

**Review by AmanKAggarwal — APPROVED** [review]
great work

**Pull request merged**
